### DISCUSSION TITLE
Is there a simpler way to get tiles of different size while writing a VIpsOperation?

### DISCUSSION THREAD

**Discussion by prasadbandodkar** [libvips/libvips#4279] [Q&A]
From https://www.libvips.org/API/current/extending.html : 
"The geometry hint is just a hint, an operation needs to be able to supply any size [VipsRegion](https://www.libvips.org/API/current/VipsRegion.html) on request. If you must have a certain size request, you can put a cache in the pipeline after your operation, see [vips_linecache()](https://www.libvips.org/API/current/libvips-conversion.html#vips-linecache) and [vips_tilecache()](https://www.libvips.org/API/current/libvips-conversion.html#vips-tilecache). You can also make requests to your operation ordered, see [vips_sequential()](https://www.libvips.org/API/current/libvips-conversion.html#vips-sequential)."

I am looking to write output image in sizes of 256x256. For now I am using "VIPS_DEMAND_STYLE_SMALLTILE" that gives me tiles of 128x128 size on average. Is there a simpler way to get the tiles of size I want without writing another object to hold a cache of tiles? If not, could you please expand on the idea "to put a cache in the pipeline after your operation"?

**Comment by jcupitt** [comment]
Hello @prasadbandodkar,

It should just work and you shouldn't need to consider tile size. Can you explain what you are trying to achieve? Are you trying to make tiles to pass on to some other program?

**Reply by prasadbandodkar** [reply]
Hi @jcupitt,

Thank you for your quick response as always! I need to process a large image tile by tile and I need each part of that image to be 256x256. I want to use vips to handle the image and I want to process the tiles using other libraries. When I do this in the build function: 
```
if (vips_image_pipelinev(object->out,
            VIPS_DEMAND_STYLE_SMALLTILE, object->in, NULL))
        return -1;

    if (vips_image_generate(negative->out,
            vips_start_one,
            negative_generate,
            vips_stop_one,
            object->in, object))
        return -1;
```
Then within the generate function:
```
static int
object_generate(VipsRegion *out_region,
    void *vseq, void *a, void *b, gboolean *stop)
{
    /* The area of the output region we have been asked to make.
     */
    VipsRect *r = &out_region->valid;
```
the size of the rectangle is 128x128 on average. How do i make that size 256x256 or some other value?
